### PR TITLE
35 modulith commodity finder

### PR DESCRIPTION
**Pull request by jixxed**

### CODE REVIEW

File: modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/configuration/MyBatisConfiguration.java [hunk 1]
```diff
@@ -24,7 +24,7 @@ public class MyBatisConfiguration {
     public SqlSessionFactory sqlSessionFactory(DataSource dataSource) throws Exception {
         SqlSessionFactoryBean sessionFactoryBean = new SqlSessionFactoryBean();
         sessionFactoryBean.setDataSource(dataSource);
-        sessionFactoryBean.setTypeHandlers(new UuidTypeHandler(), new StringListTypeHandler(), new StringTrimmingTypeHandler());
+        sessionFactoryBean.setTypeHandlers(new TypeHandler<?>[]{new UuidTypeHandler(), new StringListTypeHandler(), new StringTrimmingTypeHandler()});
```
Comment (pveeckhout) [inline modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/configuration/MyBatisConfiguration.java]: why the array initialisation?

File: modulith/build.gradle [hunk 2]
```diff
@@ -1,40 +1,32 @@
 plugins {
     id 'java'
-    id 'org.springframework.boot' version '3.1.0' apply false
+    id 'idea'
```
Comment (pveeckhout) [inline modulith/build.gradle]: we cannot assume that in an opensource system everyone will work with intellij IDEA... build files should be IDE indepenent

File: modulith/build.gradle [hunk 3]
```diff
@@ -1,40 +1,32 @@
...
     repositories {
         mavenCentral()
     }
-
-    dependencies {
-        implementation 'de.slub-dresden:urnlib:2.0.1'
-
-        compileOnly 'org.projectlombok:lombok:1.18.28'
-    }
+    task prepareKotlinBuildScriptModel {}
```
Comment (pveeckhout) [inline modulith/build.gradle]: why add kotlin tasks

File: modulith/commodityFinder/build.gradle [hunk 4]
```diff
@@ -1,24 +1,44 @@
...
+    id 'idea'
+    id "org.gradlex.extra-java-module-info" version "1.4"
 }
-
 dependencies {
     implementation project(':util')
     implementation project(':mybatis-util')
     implementation project(':messageprocessor-lib')
-    implementation 'org.springframework.boot:spring-boot-starter'
+    implementation 'org.springframework.boot:spring-boot-starter-jdbc:3.1.0'
```
Comment (pveeckhout) [inline modulith/commodityFinder/build.gradle]: not a huge fan of all the versions in the dependencies block.

maybe move these an `ext {}` block so the are bundled ?

File: modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/application/mappers/persistence/StationMapper.java [hunk 5]
```diff
@@ -1,7 +1,6 @@
-package io.edpn.backend.modulith.commodityfinder.application.mappers.persistence;
+package io.edpn.backend.modulith.commodityfinder.application.mappers;
```
Comment (pveeckhout) [inline modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/application/mappers/persistence/StationMapper.java]: package name and folder are different.... why was this changed?

File: settings.gradle [hunk 6]
```diff
@@ -1,3 +1,2 @@
 rootProject.name = 'backend'
 include 'eddnMessageListener'
-include 'modulith'
```
Comment (pveeckhout) [inline settings.gradle]: why only remove one? shouldn't it be both or none?

File: modulith/boot/src/main/java/io/edpn/backend/modulith/EDPNApplication.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package io.edpn.backend.modulith;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class EDPNApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(EDPNApplication.class, args);
```
Comment (pveeckhout) [inline modulith/boot/src/main/java/io/edpn/backend/modulith/EDPNApplication.java]: will this setup run the modulith module when running EDPNApplication ?

**Review by pveeckhout — CHANGES_REQUESTED** [review]

File: settings.gradle (hunk 6 above)
Comment (jixxed) [reply settings.gradle]: I didn't look into this as it was outside the modulith, so I assumed it was a separate project

File: modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/configuration/MyBatisConfiguration.java (hunk 1 above)
Comment (jixxed) [reply modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/configuration/MyBatisConfiguration.java]: compilation worked, but intelliJ complained about vararg initialization and showed a red squiggly line beneath it. not exactly sure what caused it

File: modulith/boot/src/main/java/io/edpn/backend/modulith/EDPNApplication.java (hunk 7 above)
Comment (jixxed) [reply modulith/boot/src/main/java/io/edpn/backend/modulith/EDPNApplication.java]: yes, but since the build is so broken I've not gotten it to launch completely.

File: modulith/build.gradle (hunk 2 above)
Comment (jixxed) [reply modulith/build.gradle]: that's a bit extreme. this plugin does not make the project assume or require anyone to use IDEA, but I removed it, since it is not needed anymore.

File: modulith/build.gradle (hunk 3 above)
Comment (jixxed) [reply modulith/build.gradle]: It was the fix suggested online. the wrapper and prepareKotlin tasks are complained about to be missing on the subprojects by gradle. not sure what the cause is, but the fix works for now

File: modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/application/mappers/persistence/StationMapper.java (hunk 5 above)
Comment (jixxed) [reply modulith/commodityFinder/src/main/java/io/edpn/backend/modulith/commodityfinder/application/mappers/persistence/StationMapper.java]: will fix

File: settings.gradle (hunk 6 above)
Comment (pveeckhout) [reply settings.gradle]: ideally we would move these two project to a repository each, so that the monolith can be the top level project

File: modulith/commodityFinder/src/main/java/module-info.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
+module edpn.commodity {
```
Comment (pveeckhout) [inline modulith/commodityFinder/src/main/java/module-info.java]: shouldn't this be named `commodityFinder`

File: modulith/messageprocessor-lib/src/main/java/module-info.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+module edpn.lib {
```
Comment (pveeckhout) [inline modulith/messageprocessor-lib/src/main/java/module-info.java]: I would prefer to keep a reference to messageProcessor in the name, as it is not just a general lib

**Review by pveeckhout — CHANGES_REQUESTED** [review]

File: modulith/commodityFinder/src/main/java/module-info.java (hunk 8 above)
Comment (jixxed) [reply modulith/commodityFinder/src/main/java/module-info.java]: the naming convention for the modules is hard to find, but the module names don't have to match 1 to 1. `-` is also not allowed in the names. renamed to edpn.commodity.finder

File: modulith/messageprocessor-lib/src/main/java/module-info.java (hunk 9 above)
Comment (jixxed) [reply modulith/messageprocessor-lib/src/main/java/module-info.java]: renamed to edpn.messageprocessor.lib

File: modulith/commodityFinder/build.gradle (hunk 4 above)
Comment (jixxed) [reply modulith/commodityFinder/build.gradle]: moved do constraints block
